### DISCUSSION TITLE
Customizable live demos (renderers?) via codefences with "meta" tags

### DISCUSSION THREAD

**Discussion by NullVoxPopuli** [vuejs/vitepress#3676] [Q&A]
### Is your feature request related to a problem? Please describe.

Vitepress offers a lot, but it doesn't allow any live demo support from the markdown.

For example, [in my own experiments](https://limber.glimdown.com/edit?c=MQAgMglgtgRgpgJxAUQCYQC4HsEChfCgDqmAFiAFICGAblQMoDGCEADhvo1qnCAGZwAdo14YqAcwDOIRlUEh4IAK6S4qENhDoEcRhgA2ATxA7BPJFyissgoRkkAaXFWkB3OPv0gXJoeY2kvFw8IJKCbKxwGAG8%2BhA0QdyJUHDSfAhYUAB0%2BADy7BA2kgBc%2BAC0IAAGcQmVIBUA%2Bqbm0hDyrPpUIg24FdXxvKw6NBBwrnVNfoit7Z0iDiAdXW3iMaFYSggi3nwYiGs1vM2IPX2Hi8OjrgoeWOMgk2bTIG2Lc3ALS4wrBwO%2BT0h4Po7mtJBstnAerhKjDxAAraSHXAAHj2Vk6ewAfLgQCAABIeYELADicSgKQQAEIUQB6NFLLHQmH4GGVeGIv5DOAjMa4aDWBDRADCmWstkE0XSmRAAHIAALiMkUmmWMV2GUAbj5Vhw0QA3hoEF0ANZqEAAX34GSgsoVSsQdKNjGNK012oF%2BpANgtVul8rgsAdUG4ED4owQbtwcAAHh6tHA%2BFQlPpooxOpJpATPFgiDh9OoY3szNIRTrxfqcSA5Rgnab1FwlBKQABeEAABi1lbazADdhbIAAFABKFuYgIQSRZBtNgDUrYAjJ3caiAwy4NjccvWJiAJobECkWhBOLOs0YQIKJQYbDyPV688TqcbCXmy0YaCpLLImnb-CbkDIjAV43iAd7egARGmEDOuB46Tt2OgpC%2B5qYkKJ7Gt%2BQHXjYG4AXSq4YuuuDmkylS4EAA&format=glimdown), I use the `live` tag to denote something should be rendered:
~~~md

markdown code here

```jsx live
export default function App() {
  return <h1>Hello World</h1>
}
```

~~~

Closest thing I've seen is https://vitepress.dev/reference/default-theme-search#custom-content-renderer
but that would require emitting a script tag in the HTML -- which... is maybe the best we can do with SSG?

### Describe the solution you'd like

have a way to hook in to code fences, with a `live` meta tag, and have them render.

I've done some thinking over here: https://github.com/NullVoxPopuli/limber/issues/1632,
the return value could be a VueComponent, since VitePress is in Vue .. but i guess.. is there any Vue left over at run time?

maybe a script tag is the only way to move forward :think

### Describe alternatives you've considered

1. I use existing behavior with script tag
  - except, I probably need to figure out how to get a compiler and a whole mess of dependencies into the build as well hmm
2. I write a competitor to VitePress, (wish me luck lol (and a time machine))

### Additional context

thanks! vitepress is great!

### Validations

- [X] Follow our [Code of Conduct](https://vuejs.org/about/coc.html)
- [X] Read the [docs](https://vitepress.dev).
- [X] Read the [Contributing Guidelines](https://github.com/vuejs/vitepress/blob/main/.github/contributing.md).
- [X] Check that there isn't already an issue that asks for the same feature to avoid creating a duplicate.

**Comment by brc-dd** [comment]
Write a markdown-it plugin (add it via markdown.config) and return a vue component from it. Let me know if you need help regarding the mdit part (it'll be ~5 line code IG).

More context on vue part - Say you register a component named `CodeLive` globally (via theme.enhanceApp), then you just need to ``return `<CodeLive :content="${stringified_content}"/>` `` for fence blocks.

**Reply by NullVoxPopuli** [reply]
oooo nice thanks!!!

**Reply by NullVoxPopuli** [reply]
I've opened this for tracking, thank you! https://github.com/NullVoxPopuli/limber/issues/1714

**Comment by pearmini** [comment]
@NullVoxPopuli 

I've created a custom theme called [genji-theme-vitepress](https://github.com/pearmini/genji) to achieve a similar effect:

````md
```js eval code=false
size = Inputs.range([50, 300], { label: "size", value: 100, step: 1 });
```

```js eval
(() => {
  const div = document.createElement("div");
  div.style.width = size + "px";
  div.style.height = "100px";
  div.style.background = "orange";
  return div;
})();
```
````

This will produce:

![Mar-29-2024 19-37-59](https://github.com/vuejs/vitepress/assets/49330279/44be229f-9e6e-47a0-94a2-a912984588db)

I'd love for you to try it out and share your feedback with me!
